tests/lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,275 @@
+      code: `
+        <script setup>
+          import { bar } from 'foo'
+
+          console.log('test1')
+          defineEmits(['update:test'])
+          console.log('test2')
+          defineProps({
+            test: Boolean
+          })
+          console.log('test3')
+        </script>
+      `,
+      output: `
+        <script setup>
+          import { bar } from 'foo'
+
+          defineProps({
+            test: Boolean
+          })
+          defineEmits(['update:test'])
+          console.log('test1')
+          console.log('test2')
+          console.log('test3')
+        </script>
+      `,
+      options: optionsPropsFirst,
+      errors: [
+        {
+          message: 'defineProps must be on top.',
+          line: 8
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          import { bar } from 'foo'
+
+          defineEmits(['update:test'])
+          defineProps({
+            test: Boolean
+          })
+
+          console.log('test1')
+        </script>
+      `,
+      output: `
+        <script setup>
+          import { bar } from 'foo'
+
+          defineProps({
+            test: Boolean
+          })
+          defineEmits(['update:test'])
+
+          console.log('test1')
+        </script>
+      `,
+      options: optionsPropsFirst,
+      errors: [
+        {
+          message: 'defineEmits must be on top.',
+          line: 6
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          console.log('test1')
+          const props = defineProps({
+            test: Boolean
+          })
+          console.log('test2')
+          const emit = defineEmits(['update:test'])
+          console.log('test3')
+        </script>
+      `,
+      output: `
+        <script setup>
+          const emit = defineEmits(['update:test'])
+          const props = defineProps({
+            test: Boolean
+          })
+          console.log('test1')
+          console.log('test2')
+          console.log('test3')
+        </script>
+      `,
+      options: optionsEmitsFirst,
+      errors: [
+        {
+          message: 'defineEmits must be on top.',
+          line: 8
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script lang="ts" setup>
+          interface Props {
+            msg?: string
+            labels?: string[]
+          }
+
+          const props = withDefaults(defineProps<Props>(), {
+            msg: 'hello',
+            labels: () => ['one', 'two']
+          })
+          const emit = defineEmits<{(e: 'update:test'): void}>()
+        </script>
+      `,
+      output: `
+        <script lang="ts" setup>
+          const emit = defineEmits<{(e: 'update:test'): void}>()
+          const props = withDefaults(defineProps<Props>(), {
+            msg: 'hello',
+            labels: () => ['one', 'two']
+          })
+          interface Props {
```

Comment:
For now `defineProps` and `defineEmits` moves on the top.
Maybe it would be convenient to move it after top import statements?

Reply:
Yes, I think `interface` and `type` statements should be allowed before `defineProps` and `defineEmits`, just like `import` statements are. The auto-fix should also move `defineProps` and `defineEmits` below all (consecutive) `import`, `interface` and `type` statements.

Consecutive meaning that it should be moved here:
```ts
import bla from 'bla';
interface Foo {};
// <--- auto-fix should move `defineProps` here
const someOtherCode = '';
import 'foo' from 'bar'; // not idiomatic, but allowed
interface SomeOtherInterface {};
defineProps();
```

Reply:
Pushed this case(https://github.com/vuejs/eslint-plugin-vue/pull/1856/files#diff-6ebdd3c21be70e29ce94a2f399884fd220d4fe9bba358ade66b0bceb325d3102R280)